sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregators/DoubleSumAggregator.java

```diff
@@ -14,26 +14,31 @@
  * limitations under the License.
  */
 
-package io.opentelemetry.sdk.metrics;
+package io.opentelemetry.sdk.metrics.aggregators;
 
 import com.google.common.util.concurrent.AtomicDouble;
 
-final class DoubleSumAggregator implements Aggregator<DoubleSumAggregator> {
+public final class DoubleSumAggregator implements Aggregator {
   // TODO: Change to use DoubleAdder when changed to java8.
   private final AtomicDouble current;
 
-  DoubleSumAggregator() {
+  public DoubleSumAggregator() {
     current = new AtomicDouble(0.0);
   }
 
   @Override
-  public void merge(DoubleSumAggregator other) {
-    this.current.getAndAdd(other.current.get());
+  public void mergeAndReset(Aggregator aggregator) {
+    if (!(aggregator instanceof DoubleSumAggregator)) {
+      return;
+    }
+
+    DoubleSumAggregator other = (DoubleSumAggregator) aggregator;
+    other.current.getAndAdd(this.current.getAndSet(0));
   }
 
   @Override
   public void recordLong(long value) {
-    throw new UnsupportedOperationException("This Aggregator does not support long values");
+    throw new UnsupportedOperationException("This is a DoubleSumAggregator");
```

Comment:
hmm. this smells like we might not have the right class/interface structure. Can we make the recordable type a parameter of the aggregator type?

Reply:
not sure I understand the proposal, can you clarify a bit, maybe add a small example.

Reply:
If all the aggegrators need both methods on them, and all of them throw an UOE with one of them, that seems like the interface isn't really helping us out.  

If we had an Aggregator<T>, then you could just have a record(T value) on that interface, and each implementation would only have to have one record method, rather than two.

DoubleSumAggregator implements Aggregator<Double> then.

Reply:
I do wonder if the interface ends up being some extra baggage to carry around, though. I'll keep an eye on this as things evolve.